packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -957,30 +958,45 @@ function getRcFilename() {
   return DEFAULT_RC_FILENAME as Filename;
 }
 
+async function tryRead(p: PortablePath) {
+  try {
+    return xfs.readFilePromise(p);
+  } catch {
+    return Buffer.of();
+  }
+}
+
+async function isSameBinaryContent(a: PortablePath, b: PortablePath) {
+  return Buffer.compare(...await Promise.all([
+    tryRead(a),
+    tryRead(b),
+  ])) === 0;
+}
+
+async function isSameBinaryInode(a: PortablePath, b: PortablePath) {
+  const [aStat, bStat] = await Promise.all([
+    xfs.statPromise(a),
+    xfs.statPromise(b),
+  ]);
+
+  return aStat.dev === bStat.dev && aStat.ino === bStat.ino;
+}
+
+const isSameBinary = process.platform === `win32`
+  ? isSameBinaryContent
+  : isSameBinaryInode;
```

Comment:
This wont catch cases where the file content is the same but he inode isn't, for example with Corepack and `yarnPath`.

Reply:
Having exactly the same version in Corepack and yarnPath (considering that V4 use Corepack without yarnPath when possible) seems in theory unlikely enough that it's tolerable

Perhaps a problem though is that existing projects will have a hard time noticing that they should remove `yarnPath` from their configuration and fully migrate to Corepack... 🤔

Reply:
I'll make a separate PR to add a warning when `yarnPath` is detected when it could be Corepack